File: src/Other/Internals/Search/SamplesInfoLoader.cs
using OpenSilver.Samples.Showcase.Search;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection;
using Microsoft.Maui.Devices;

namespace OpenSilver.Samples.Showcase.Search
{
    public static class SamplesInfoLoader
    {
        //keep a mapping of the name to the type, so we can later instantiate the type after a search.
        private static Dictionary<string, Type> controlTypeMap = new Dictionary<string, Type>();

        public static List<SearchableItem> GetAllControls()
        {
            var controls = new List<SearchableItem>();

            var types = Assembly.GetExecutingAssembly()
                                .GetTypes()
                                .Where(t => t.IsClass && t.GetCustomAttribute<SearchKeywordsAttribute>() != null);

            foreach (var type in types)
            {
                var attribute = type.GetCustomAttribute<SearchKeywordsAttribute>();
                var name = type.Name;
                bool ignoreType = DeviceInfo.Current.Platform == DevicePlatform.Unknown && attribute?.Keywords.Contains("maui") == true;
                if (!ignoreType)
                controls.Add(new SearchableItem
                {
                    Name = name,
                    Keywords = attribute?.Keywords.ToList() ?? new List<string>()
                });
                controlTypeMap[name] = type;
            }

            return controls;
        }

        public static Type GetControlTypeByName(string name)
        {
            return controlTypeMap.TryGetValue(name, out var type) ? type : null;
        }
    }
}

File: src/Other/Internals/Search/SamplesInfoLoader.vb
Imports OpenSilver.Samples.Showcase.Search
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Reflection
Imports Microsoft.Maui.Devices

Namespace OpenSilver.Samples.Showcase.Search
    Public Module SamplesInfoLoader
        ' Keep a mapping of the name to the type, so we can later instantiate the type after a search.
        Private ReadOnly controlTypeMap As New Dictionary(Of String, Type)()

        Public Function GetAllControls() As List(Of SearchableItem)
            Dim controls As New List(Of SearchableItem)()

            Dim types = Assembly.GetExecutingAssembly().
                        GetTypes().
                        Where(Function(t) t.IsClass AndAlso t.GetCustomAttribute(Of SearchKeywordsAttribute)() IsNot Nothing)

            For Each type In types
                Dim attribute = type.GetCustomAttribute(Of SearchKeywordsAttribute)()
                Dim name As String = type.Name
                Dim ignoreType As Boolean = DeviceInfo.Current.Platform = DevicePlatform.Unknown AndAlso
                                            attribute?.Keywords.Contains("maui") = True

                If Not ignoreType Then
                    controls.Add(New SearchableItem With {
                        .Name = name,
                        .Keywords = If(attribute?.Keywords.ToList(), New List(Of String)())
                    })
                End If

                controlTypeMap(name) = type
            Next

            Return controls
        End Function

        Public Function GetControlTypeByName(name As String) As Type
            Dim type As Type = Nothing
            Return If(controlTypeMap.TryGetValue(name, type), type, Nothing)
        End Function
    End Module
End Namespace
